core/domain/rights_manager.py

```diff
@@ -753,7 +774,15 @@ def _assign_role(
     committer_id = committer.user_id
     activity_rights = _get_activity_rights(activity_type, activity_id)
 
-    if not check_can_modify_activity_roles(committer, activity_rights):
+    role_is_voice_artist = new_role == constants.ROLE_VOICE_ARTIST
+    user_can_assigning_voice_artist = check_can_manage_voice_artist_in_activity(
+        committer, activity_rights)
+
+    user_can_assigning_role = (
+        check_can_modify_activity_roles(committer, activity_rights)) or (
```

Comment:
Does "activity roles" here include voice artists?

It probably shouldn't, given the logic here, but it is a bit confusing that way. Maybe the method name should be changed to check_can_modify_core_activity_roles (to convey a sense of just 'owner'/'editor'/etc.)